apogee/tools/read.py

```diff
@@ -571,19 +583,43 @@ def rcsample(main=False,dr=None,xmatch=None,
         astroNNdata= astroNNDistances()
         # Match on (ra,dec)
         m1,m2,_= _xmatch(data,astroNNdata,maxdist=2.,
-            colRA1='RA',colDec1='DEC',colRA2='ra_apogee',colDec2='dec_apogee')
+            colRA1='RA',colDec1='DEC', colRA2='ra_apogee', colDec2='dec_apogee')
         data= data[m1]
         astroNNdata= astroNNdata[m2]
         data= _add_astroNN_distances(data,astroNNdata)
     if use_astroNN or kwargs.get('astroNN',False) or use_astroNN_ages:
         _warn_astroNN_ages()
-        astroNNdata= astroNNAges()
         # Match on (ra,dec)
-        m1,m2,_= _xmatch(data,astroNNdata,maxdist=2.,
-            colRA1='RA',colDec1='DEC',colRA2='ra_apogee',colDec2='dec_apogee')
-        data= data[m1]
-        astroNNdata= astroNNdata[m2]
-        data= _add_astroNN_ages(data,astroNNdata)
+        if int(dr) < 16:
+            astroNNdata= astroNNAges()
+            data = _add_astroNN_ages(data,astroNNdata)
+        else:
+            astroNNdata = astroNN() #ages are in the main VAC for DR > 16
+            m1,m2,_= _xmatch(data,astroNNdata,maxdist=2.,
+                colRA1='RA',colDec1='DEC',
+                             colRA2='ra_apogee',
+                             colDec2='dec_apogee')
+            data= data[m1]
+            astroNNdata= astroNNdata[m2]
+            data= _add_astroNN_ages(data,astroNNdata)
```

Comment:
Hi Jo, I did this because when testing I realised that the DR14 Ages catalogue has no RA/DEC, so the match as it was didn't work (at least for me). The ID based match that `_add_astroNN_ages` does is ok in this case, I think? I may be wrong, but this seemed to work when I tested it.

for DR16, the match as before works fine because it uses the full VAC (as you mentioned).

Reply:
I see, okay. But is it necessary to do the ``xmatch`` at all then, if there is an ID-based match after? I know it's a detail...

Reply:
ah I see what you mean, yes, that is pretty dumb of me.... will switch that up now

Reply:
ah - no I remember now: in the case of DR16 `_add_astroNN_ages` just assumes a line-by-line match, so we need the xmatch by RA and Dec for the RC sample after loading with `astroNN()`